qiskit_experiments/library/characterization/multi_state_discrimination.py

```diff
@@ -0,0 +1,141 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Multi state discrimination experiment."""
+
+from typing import Dict, List, Optional
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Gate, Parameter
+from qiskit.providers import Backend
+from qiskit.providers.options import Options
+from qiskit.pulse import ScheduleBlock
+from qiskit.qobj.utils import MeasLevel, MeasReturnType
+from qiskit_experiments.framework import BaseExperiment
+from qiskit_experiments.library.characterization import MultiStateDiscriminationAnalysis
+
+
+class MultiStateDiscrimination(BaseExperiment):
+    r"""An experiment that discriminates between the first :math:`n` energy states.
+
+    # section: overview
+
+        The experiment creates :math:`n` circuits that prepare, respectively, the energy states
+        :math:`|0\rangle,\cdots,|n-1\rangle`. For, e.g., :math:`n=4` the circuits are of the form
+
+        .. parsed-literal::
+
+            Circuit preparing :math:`|0\rangle`
+
+                       ░ ┌─┐
+                   q: ─░─┤M├
+                       ░ └╥┘
+                meas: ════╩═
+
+            ...
+
+            Circuit preparing :math:`|3\rangle`
+
+                      ┌───┐┌─────┐┌─────┐ ░ ┌─┐
+                   q: ┤ X ├┤ x12 ├┤ x23 ├─░─┤M├
+                      └───┘└─────┘└─────┘ ░ └╥┘
+                meas: ═══════════════════════╩═
+
+    # section: reference
+        `Qiskit Textbook <https://qiskit.org/textbook/ch-quantum-hardware/accessing\
+        _higher_energy_states.html>`_.
+
+    # section: tutorial
+        :doc:`TODO?`
+    """
+
+    @classmethod
+    def _default_run_options(cls) -> Options:
+        """Default option values for the experiment :meth:`run` method."""
+        options = super()._default_run_options()
+
+        options.meas_level = MeasLevel.KERNELED
+        options.meas_return = MeasReturnType.SINGLE
+
+        return options
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default values for the number of states if none is given.
+
+        Experiment Options:
+            n_states (int): The number of states to discriminate.
+
+        """
+        options = super()._default_experiment_options()
+        options.n_states = 2
```

Comment:
```suggestion
        options.n_states = 2
        schedules = None
```

Reply:
do you mean` options.schedules = None` ?

Reply:
Ah yes.